### PR TITLE
Added TravelPackage controller, model, repository and service.

### PR DESCRIPTION
**Pull request by dragomirescuionut**

### CODE REVIEW

**Comment by Catalinn2** [comment]
Ok

File: src/main/java/com/teamtiger/travelbookingsys/controllers/TravelPackageController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.teamtiger.travelbookingsys.controllers;
+
```
Comment (Catalinn2) [inline src/main/java/com/teamtiger/travelbookingsys/controllers/TravelPackageController.java:2]: Super

File: src/main/java/com/teamtiger/travelbookingsys/controllers/TravelPackageController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public class TravelPackageController {
+    private final TravelPackageService travelPackageService;
+
+    public TravelPackageController(TravelPackageService travelPackageService) {
+        this.travelPackageService = travelPackageService;
+    }
+
+    @PostMapping
+    public ResponseEntity<TravelPackageDTO> createTravelPackage(@RequestBody TravelPackageDTO travelPackageDTO) {
+        return ResponseEntity.ok(travelPackageService.createPackage(travelPackageDTO));
```
Comment (AlexandruIliescu) [inline src/main/java/com/teamtiger/travelbookingsys/controllers/TravelPackageController.java:22]: If we have controller method `createTravelPackage`, why should we have service method `createPackage`?
Please consider consistency when choosing method names.

File: src/main/java/com/teamtiger/travelbookingsys/models/dtos/TravelPackageDTO.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.teamtiger.travelbookingsys.models.dtos;
+
+import lombok.Data;
+
+@Data
+public class TravelPackageDTO {
+
+    private Long id;
+
```
Comment (AlexandruIliescu) [inline src/main/java/com/teamtiger/travelbookingsys/models/dtos/TravelPackageDTO.java:9]: Please don't leave empty rows where we don't need them. In this case, we can delete the empty row from line 9.
Please don't forget about code formatting.

File: src/main/java/com/teamtiger/travelbookingsys/models/dtos/TravelPackageDTO.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    private String packageName;
+    private String description;
+    private String destination;
+    private int duration;
+    private double pricePerPerson;
```
Comment (AlexandruIliescu) [inline src/main/java/com/teamtiger/travelbookingsys/models/dtos/TravelPackageDTO.java:14]: Can we maybe find a better name for this property? Can we maybe call it just `price`? Would that make sense?

File: src/main/java/com/teamtiger/travelbookingsys/models/entities/TravelPackage.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    @Column(name = "package_name")
+    private String packageName;
+    @Column(name = "description")
+    private String description;
+    @Column(name = "destination")
+    private String destination;
+    @Column(name = "duration")
+    private int duration;
+    @Column(name = "price_per_person")
+    private double pricePerPerson;
```
Comment (AlexandruIliescu) [inline src/main/java/com/teamtiger/travelbookingsys/models/entities/TravelPackage.java:22]: If you will update `pricePerPerson` property in the DTO, please don't forget to also update it here.

File: src/main/java/com/teamtiger/travelbookingsys/services/TravelPackageService.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.teamtiger.travelbookingsys.services;
+
+import com.teamtiger.travelbookingsys.models.dtos.TravelPackageDTO;
+
+public interface TravelPackageService {
+    TravelPackageDTO createPackage(TravelPackageDTO travelPackageDTO);
+}
+
```
Comment (AlexandruIliescu) [inline src/main/java/com/teamtiger/travelbookingsys/services/TravelPackageService.java:8]: Please don't forget empty rows that we don't need.

File: src/main/java/com/teamtiger/travelbookingsys/services/TravelPackageServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+        TravelPackageDTO travelPackageResponseDTO = new TravelPackageDTO();
+        travelPackageResponseDTO.setId(savedTravelPackage.getId());
+        travelPackageResponseDTO.setPackageName(savedTravelPackage.getPackageName());
+        travelPackageResponseDTO.setDescription(savedTravelPackage.getDescription());
+        travelPackageResponseDTO.setDestination(savedTravelPackage.getDestination());
+        travelPackageResponseDTO.setDuration(savedTravelPackage.getDuration());
+        travelPackageResponseDTO.setPricePerPerson(savedTravelPackage.getPricePerPerson());
+
+        return travelPackageResponseDTO;
```
Comment (AlexandruIliescu) [inline src/main/java/com/teamtiger/travelbookingsys/services/TravelPackageServiceImpl.java:38]: Can we improve this method by extracting the conversions to other methods? The logic for entity - dto conversion and the other way around can be moved to distinct methods, and we can end up having a slimmer service method.

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -10,3 +10,4 @@ spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
 # Logging properties
 logging.level.org.hibernate.SQL=DEBUG
 logging.level.org.hibernate.type.descriptor.sql.BasicBinder=TRACE
+spring.main.allow-bean-definition-overriding=true
```
Comment (AlexandruIliescu) [inline src/main/resources/application.properties:13]: Do we need this? Do we have a particular reason to add this property?
At this moment we don't need that property.

**Review by Catalinn2 — APPROVED** [review]

**Comment by Catalinn2** [comment]
Approved

**Pull request merged**